### ISSUE TITLE
Allow branches to be build independently

### ISSUE THREAD

**Issue by martinheidegger** [Strider-CD/strider#575]
Right now branches are built in the same queue: branch a is built after branch a has finished. I assume that is the case because of some operation integrity.

This feature made strider useless for me.

I wanted to use strider to have experiments run on changes in git. Each experiment would take hours in time so I just wanted to make slight variants and see if the computer behaves differently. Like it is now I can not run this in parallel and instead of running 200 tests in an hour time i need to wait 200 hours. 

Long story short: If I were able to separate the branches to run independently I could setup a branch for each of the 200 steps and had them run one by one.
